pkg/apis/middleware/session.go

```diff
@@ -40,7 +40,7 @@ func CreateTokenToSessionFunc(verify VerifyFunc) TokenToSessionFunc {
 			claims.Email = claims.Subject
 		}
 
-		if claims.Verified != nil && !*claims.Verified {
+		if claims.Verified != nil && !*claims.Verified && !allowUnverifiedEmail {
 			return nil, fmt.Errorf("email in id_token (%s) isn't verified", claims.Email)
 		}
```

Comment:
Therefore I think this if block should be moved to the `Verify()` in `pkg > providers > oidc > verifier.go` where it belongs.

Reply:
Unfortunately, that means you will have to refactor your tests again.